transformer_engine/pytorch/attention.py

```diff
@@ -2643,6 +2646,28 @@ def forward(
         if qkv_format is None:
             qkv_format = self.qkv_format
 
+        if inference_params is not None:
+            assert qkv_format == "sbhd"
```

Comment:
Why do we need to assume that? Shouldn't we actually support all cases here (maybe apart from thd, since that would be much harder so we should not tackle it in this PR)? We could inform users that `sbhd` would give the best performance, but `bshd` should also work.